checker/manual/constant-value-checker.tex

```diff
@@ -185,22 +224,27 @@ \section{Warnings\label{value-checker-warnings}}
 \end{itemize}
 \end{sloppypar}
 
-There is one other situation in which the Constant Value Checker produces a
+There are some other situations in which the Constant Value Checker produces a
 warning message:
 
 \begin{sloppypar}
 \begin{itemize}
-\item \code{[too.many.values.given] Annotation ignored because the maximum
-  number of values tracked is 10.}
+\item \code{[too.many.values.given] The maximum number of arguments permitted is 10.}
 
   The Constant Value Checker only tracks up to 10 possible values for an
   expression.  If you write an annotation with more values than will be
-  tracked, the annotation is ignored.
+  tracked, the annotation is replaced with \<@IntRange> or \<@UnknownVal>.
 
-\end{itemize}
-\end{sloppypar}
+\item \code{[too.narrow.int.range] The smallest interval of integer range permitted is 10.}
```

Comment:
`too.narrow.int.range` should not be a warning.
We don't want to forbid users from writing
 @IntRange(from=0, to=9)
It would be perverse to instead force them to write
 @IntRange({0, 1, 2, 3, 4, 5, 6, 7, 8, 9})

At most, you should silently change the annotation.  However, I think it's better not to change the annotation, but to keep the annotation as written.

Every check for @IntVal annotations should work identically when there is an @IntRange annotation whose range encompasses 10 or fewer values.  If this pull request doesn't do that, then please update it to do so, and add tests of that functionality.